cluster-autoscaler/utils/kubernetes/factory.go

```diff
@@ -0,0 +1,18 @@
+package kubernetes
+
+import (
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+	kube_client "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	kube_record "k8s.io/kubernetes/pkg/client/record"
+	v1core "k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/core/v1"
+
+	"github.com/golang/glog"
+)
+
+// CreateEventRecorder creates an event recorder to send custom events to Kubernetes to be recorded for targeted Kubernetes objects
+func CreateEventRecorder(kubeClient kube_client.Interface) kube_record.EventRecorder {
+	eventBroadcaster := kube_record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: kubeClient.Core().Events("")})
```

Comment:
I couldn't manage to run through this func with a fake kube client.
After digging a bit, I believe that this line relates to the source of problem observed in https://github.com/kubernetes/contrib/pull/2226#issuecomment-274432557

When `kubeClient` is a fake client, `StartRecordingToSink` ends up calling [`func (c *FakeCoreV1) RESTClient() `](https://github.com/kubernetes/kubernetes/blob/master/federation/client/clientset_generated/federation_internalclientset/typed/core/internalversion/fake/fake_core_client.go#L49-L54) which returns nil and then eventually cause a panic from nil dereference errors.